### PR TITLE
[#2222] Add support for versioning the python distribution.

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This updates the paver.sh to make use of the newly versioned python-package archives.

Changes
=======

paver.sh was updated to use the versioned python when possible... but to also fall back to the archive which is not versioned.

The paver.conf was updated to include allow explicitly defining the version to use. 

Looks like solaris10 is the only one with an older version of Python.
There is also HPUX without a version at all... but hope this will be updated soon,

How to try and test the changes
===============================

reviewers: @alibotean @brunogola 

Check that changes make sense.

@dumol let me know if you have a better idea for implemeting test_version_exists

1.paver get_python

this is used by server release process 

```
# direct get for a python which is not versioned
./paver.sh get_python hpux1131 ia64
# Direct get for python with version
./paver.sh get_python solaris11 x64
```

2.Upgrade from installed unversioned version
-------------

```
# Make sure the python looks like an old one without version
rm build-ubuntu1604-x64/lib/PYTHON_PACKAGE_VERSION
./paver.sh deps
```

### CODE REVIEW

**Comment by adiroiban** [comment]
there are still some failures due to pip tmp files ... but this should be ready for a first round of review

thanks!


needs-review

File: paver.conf [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This is here since we use the same bootstrap script for all projects.
 # Once each project uses its own bootstrap script, we can get rid of this.
 BRINK_VERSION='skip'
-PYTHON_VERSION='python2.7'
+PYTHON_CONFIGURATION='default@2.7.10.8fa7d09:solaris10-x64@2.7.8.8fa7d09'
```
Comment (adiroiban) [inline paver.conf]: Thanks.  I can then have

```
default@2.7.10.8fa7d09:solaris10@2.7.8.8fa7d09'
```

as this will match 'solaris10*'

**Comment by brunogola** [comment]
the code seems fine. 

changes-approved

**Comment by alibotean** [comment]
> # direct get for a python which is not versioned
> ./paver.sh get_python hpux1131 ia64
> # Direct get for python with version
> ./paver.sh get_python solaris11 x64

Both commands look unversioned to my untrained eye :smile:

**Comment by alibotean** [comment]
Can't get the versioned one to work. The others work as expected.

**Comment by adiroiban** [comment]
> Both commands look unversioned to my untrained eye

you should check the log to see what files is transferred .... 

http://binary.chevah.com/production/python/python2.7-hpux1131-ia64.tar.gz

vs 

http://binary.chevah.com/production/python/solaris10/sparc64/python-2.7.8.8fa7d09-solaris10-sparc64.tar.gz

**Comment by alibotean** [comment]
I see.

changes-approved

**Pull request closed without merging**